File: http-server/src/test/java/com/example/HelloServiceTest.java
package com.example;

import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.RegisterExtension;

import com.linecorp.armeria.client.WebClient;
import com.linecorp.armeria.common.AggregatedHttpResponse;
import com.linecorp.armeria.server.ServerBuilder;
import com.linecorp.armeria.testing.junit5.server.ServerExtension;

class HelloServiceTest {
    @RegisterExtension
    static final ServerExtension server = new ServerExtension() {
        @Override
        protected void configure(ServerBuilder sb) throws Exception {
            sb.annotatedService(new HelloService());
        }
    };

    @Test
    void hello1() {
        final WebClient client = client();
        final AggregatedHttpResponse response = client.get("/hello1").aggregate().join();
        Assertions.assertEquals("Hello!", response.contentUtf8());
    }

    @Test
    void hello2() {
        final WebClient client = client();
        final AggregatedHttpResponse response = client.get("/hello2").aggregate().join();
        Assertions.assertEquals("Hello!!", response.contentUtf8());
    }

    private static WebClient client() {
        return WebClient.of(server.httpUri());
    }
}

File: http-server/http-server-kotlin/src/test/kotlin/com/example/HelloServiceTest.kt
package com.example;

import com.linecorp.armeria.client.WebClient
import com.linecorp.armeria.common.QueryParams
import com.linecorp.armeria.server.ServerBuilder
import com.linecorp.armeria.testing.junit5.server.ServerExtension
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.RegisterExtension

class HelloServiceTest {
    @RegisterExtension
    val server: ServerExtension = object : ServerExtension() {
        override fun configure(sb: ServerBuilder) {
            sb.annotatedService(HelloService());
        }

        override fun runForEachTest(): Boolean {
            return true
        }
    }

    @Test
    fun hello1() {
        val response = client().get("/hello1").aggregate().join();
        Assertions.assertEquals("{\"message\":\"Hello!\"}", response.contentUtf8());
    }

    @Test
    fun hello2() {
        val response = client().get("/hello2", QueryParams.of("name", "test")).aggregate().join();
        Assertions.assertEquals("{\"message\":\"Hello, test!\"}", response.contentUtf8());
    }

    private fun client(): WebClient = WebClient.of(server.httpUri());
}
